js/ui/control/fullscreen_control.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+const DOM = require('../../util/dom');
+const util = require('../../util/util');
+const window = require('../../util/window');
+
+/**
+ * A `FullscreenControl` control contains a button for toggling the map in and out of fullscreen mode.
+ *
+ * @implements {IControl}
+ * @example
+ * map.addControl(new mapboxgl.FullscreenControl());
+ * @see [View a fullscreen map](https://www.mapbox.com/mapbox-gl-js/example/fullscreen/)
+ */
+
+class FullscreenControl {
+
+    constructor() {
+        this._fullscreen = false;
+        util.bindAll([
+            '_onClickFullscreen',
+            '_changeIcon'
+        ], this);
+    }
+
+    onAdd(map) {
+        const className = 'mapboxgl-ctrl';
+        const container = this._container = DOM.create('div', `${className} mapboxgl-ctrl-group`);
+        const button = this._fullscreenButton = DOM.create('button', (`${className}-icon ${className}-fullscreen`), this._container);
+        button.setAttribute("aria-label", "Toggle fullscreen");
+        this._fullscreenButton.addEventListener('click', this._onClickFullscreen);
+        this._mapContainer = map.getContainer();
+        window.document.addEventListener('webkitfullscreenchange', this._changeIcon);
+        return container;
+    }
+
+    onRemove() {
+        this._container.parentNode.removeChild(this._container);
+        this._map = null;
+    }
+
+    _isFullscreen() {
+        return this._fullscreen;
+    }
+
+    _changeIcon() {
```

Comment:
`_changeIcon`, as the event handler bound to fullscreenchange event, needs to check if the event corresponds to the appropriate page element. There might be more than one map on the page, or other elements that go fullscreen. [Here's the equivalent code in the Leaflet plugin.](https://github.com/Leaflet/Leaflet.fullscreen/blob/0c599cf4f369ee50658dc6a77b6082e239fd4f9c/dist/Leaflet.fullscreen.js#L99-L111)